### PR TITLE
Kaixiang fix volunteering time

### PR DESCRIPTION
**Pull request by KaixiangGU**
This PR is about Fixing volunteering hours by category. Changes were made in this PR:

1. Added three new categories: Total Intangible Hours, Total Economics Hours, and Total Stewardship Hours.
2. Total Intangible Hours should pull back all your past Intangible hours.
3. All those Category hours should be increased based on your selected project in your time entry, as well as "Total Tangible Hours" should be increased at the same time.
4. When someone's hours are manually edited, new hours should be added on to them and hours should keep tracking afterwards. 
5. When an old time entry was edited, especially volunteering times were edited, category hours and total Total Tangible Hours should increase/decrease depending on your old hours.
6. All hours should be displayed in two decimal places.

### CODE REVIEW

File: src/actions/timeEntries.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const editTimeEntry = (timeEntryId, timeEntry, oldDateOfWork) => {
     try {
       const res = await axios.put(url, timeEntry);
       dispatch(updateTimeEntries(timeEntry, oldDateOfWork));
-      return res.status;
+      return res.status
```
Comment (LeeSingChanYau) [inline src/actions/timeEntries.js:76]: Did you delete the semicolon on purpose?

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx [hunk 2]
```diff
@@ -75,13 +75,14 @@ const TimeEntryForm = (props) => {
...
   const [tasks, setTasks] = useState([]);
+  const [formDataBeforeEdit, setFormDataBeforeEdit] = useState({});
 
   const fromTimer = !_.isEmpty(timer);
   const { userProfile, currentUserRole } = useSelector(getTimeEntryFormData);
 
   const dispatch = useDispatch();
 
-  const tangibleInfoToggle = (e) => {
+  const tangibleInfoToggle = e => {
```
Comment (LeeSingChanYau) [inline src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:85]: Did you delete the parenthesis in line 85 and 47 to be consistent in how you write the code?

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx [hunk 3]
```diff
@@ -112,29 +113,38 @@ const TimeEntryForm = (props) => {
...
       })
       .catch(err => console.log(err));
   }, []);
 
+  //grab form data before editing
+  useEffect(() => {
+    if (isOpen && edit) {
+      setFormDataBeforeEdit(inputs);
+    } else {
+      setFormDataBeforeEdit({});
```
Comment (LeeSingChanYau) [inline src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:127]: is it necessary to set setFormDataBefore to empty object every time that isOpen or edit is false?

**Review by LeeSingChanYau — COMMENTED** [review]
I do not see any problems with the code, just have some doubts about the syntax.

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx (hunk 2 above)
Comment (KaixiangGU) [reply src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:85]: I think this is caused by the prettier auto re-format the file when I changed the code on this file.

File: src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx (hunk 3 above)
Comment (KaixiangGU) [reply src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx:127]: Thanks for pointing it out, it's not necessary to write else condition, I'm gonna change it.

File: src/actions/timeEntries.js (hunk 1 above)
Comment (KaixiangGU) [reply src/actions/timeEntries.js:76]: Hmm, no, maybe I accidentally deleted it.

**Comment by guiwustro** [comment]
Its working well here. The only issue that I found is that I need to refresh the page to see the changes.

**Review by guiwustro — APPROVED** [review]
Its working well here. The only issue that I found is that I need to refresh the page to see the changes. I uploaded a video in the thread of the PR at slack

**Pull request merged**